tests/Cache/CacheTest.php

```diff
@@ -194,6 +194,25 @@ public function provideCanConvertToAndFromJsonCases()
         );
     }
 
+    public function testToJsonThrowsExceptionOnInvalid()
+    {
+        $this->setExpectedException(
+            'UnexpectedValueException',
+            'Can not encode cache signature to JSON, error: "Malformed UTF-8 characters, possibly incorrectly encoded". If you have non-UTF8 chars in your signature, like in license for `header_comment`, consider enabling `ext-mbstring` or install `symfony/polyfill-mbstring`.'
+        );
+
+        $invalidUtf8Sequence = "\xB1\x31";
+
+        $signature = $this->prophesize('PhpCsFixer\Cache\SignatureInterface');
+        $signature->getPhpVersion()->willReturn($invalidUtf8Sequence);
```

Comment:
I would rather use the invalid UTF-8 sequence in `getRules()` as it is more likely to return such value.